src/routes/tools/_common/_helpers.js

```diff
@@ -0,0 +1,152 @@
+import { rollup, sort, group } from "d3-array";
+
+/**
+ * Groups data for 2 or more timeseries by year, outputs a single timeseries with
+ * min and max value for each year.
+ * Used for creating envelopes to represent an ensemble range.
+ * This data structure is used as an input to the d3 area function
+ * @param {array} _data
+ * @return {object}
+ */
+export function buildEnvelope(_data) {
+  const dataMap = rollup(
```

Comment:
just FYI you can also use `d3.rollups` (note the additional `s`) if you want a 2D array data structure instead of a Map.

Reply:
Fixed via https://github.com/berkeley-gif/caladapt-website-2021/commit/808662890308737fcc132ea47d23b6019e7a8890